middle_end/flambda/from_lambda/cps_conversion.ml

```diff
@@ -1028,10 +1028,15 @@ and cps_switch (switch : L.lambda_switch) ~scrutinee (k : Continuation.t)
   in
   let build_switch scrutinee wrappers =
     let const_switch = I.Switch (scrutinee, const_switch) in
-    let block_switch = cps_non_tail
-      (L.Lprim (Pgettag, [L.Lvar scrutinee], Loc_unknown))
-      (fun scrutinee -> I.Switch (scrutinee, block_switch))
-      k_exn
+    let scrutinee_tag = Ident.create_local "scrutinee_tag" in
+    let block_switch = I.Let ( scrutinee_tag,
```

Comment:
Nitpick: if your `let` binding takes several lines, I prefer keeping only the pattern on the first line (introducing a line break just after the `=`).